kernel/gic/src/redist_interface.rs

```diff
@@ -0,0 +1,93 @@
+//! Redistributor Interface
+//!
+//! Included functionnality:
+//! - Initializing the interface
+//! - Enabling or disabling the forwarding of PPIs & SGIs based on their numbers
+
+use super::GicMappedPage;
+use super::U32BYTES;
+use super::IntNumber;
+use super::Enabled;
+
+mod offset {
+    use super::U32BYTES;
+    pub const RD_WAKER: usize = 0x14 / U32BYTES;
+    pub const IGROUPR:  usize = 0x80 / U32BYTES;
+    pub const SGI_ISENABLER: usize = 0x100 / U32BYTES;
+    pub const SGI_ICENABLER: usize = 0x180 / U32BYTES;
+}
+
+const RD_WAKER_PROCESSOR_SLEEP: u32 = 1 << 1;
+const RD_WAKER_CHLIDREN_ASLEEP: u32 = 1 << 2;
+
+// const GROUP_0: u32 = 0;
+const GROUP_1: u32 = 1;
+
+// This timeout value works on some ARM SoCs:
+// - qemu's virt virtual machine
+//
+// (if the value works for your SoC, please add it to this list.)
+//
+// If the redistributor's initialization times out, it means either:
+// - that your ARM SoC is not GICv3 compliant (try initializing it as GICv2)
+// - that the timeout value is too low for your ARM SoC. Try increasing it
+// to see if the booting sequence continues.
+//
+// If it wasn't enough for your machine, reach out to the Theseus
+// developers (or directly submit a PR).
+const TIMEOUT_ITERATIONS: usize = 0x10_000;
+
+/// Initializes the redistributor by waking
+/// it up and checking that it's awake
+pub fn init(registers: &mut GicMappedPage) -> Result<(), &'static str> {
+    let mut reg;
+    reg = registers.read_volatile(offset::RD_WAKER);
+
+    // Wake the redistributor
+    reg &= !RD_WAKER_PROCESSOR_SLEEP;
+
+    registers.write_volatile(offset::RD_WAKER, reg);
+
+    // then poll ChildrenAsleep until it's cleared
+
+    let children_asleep = || {
+        registers.read_volatile(offset::RD_WAKER) & RD_WAKER_CHLIDREN_ASLEEP > 0
+    };
+
+    let mut counter = 0;
+    let mut timed_out = || {
+        counter += 1;
+        counter >= TIMEOUT_ITERATIONS
+    };
+
+    while children_asleep() && !timed_out() { }
```

Comment:
so we use this timeout here but not elsewhere? (it was commented out in the other file).

Shouldn't we either use it in all relevant places, or never at all?

Reply:
Which other file? The redistributor is the only component that has such polling initialization requirements.